python/paddle/distributed/fleet/meta_optimizers/sharding_optimizer.py

```diff
@@ -309,14 +309,15 @@ def minimize_impl(self,
                     use_calc_stream=True)
             if self.hybrid_dp and self.hybrid_dp_mode == "pp_hybrid_dp":
                 first_optimize_op_index = get_first_check_finite_and_unscale_op_idx(
-                    main_block)
-                insert_allreduce_ops(
-                    main_block,
-                    first_optimize_op_index,
-                    self.dp_ring_id,
-                    accumulated_grad_names,
-                    core.op_proto_and_checker_maker.OpRole.Optimize,
-                    use_calc_stream=True)
+                    main_block, raise_error=self.user_defined_strategy.amp)
```

Comment:
what is the reason for this modification ? for npu hang bug?

Reply:
To solove nup hang bugs!